### PR TITLE
Prevent flashing preview

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #4570

### Summary of Changes

The preview popup can go crazy when the debugger is struggling w/ a big bundle or other computations like mapping scopes... This makes it much much simpler.

- remove redundant renders (caused by empty symbols obj)
- move getInScopeLocations to an action so it's only computed once
- remove parser getSymbols call, which was async and went over the wire
- moved utils/expression/updatePreview to an action
- some other stuff as well... 

We might need to break this up to make it easier to land...

#### Next Steps

- i don't think we need 2 actions, one should do...

### Test Plan

- Just manual testing
- I'd love to add more unit tests

### CODE REVIEW

File: src/components/Editor/Preview/index.js [hunk 1]
```diff
@@ -126,21 +127,20 @@ class Preview extends PureComponent {
...
-  setPreview,
+  updatePreview,
   clearPreview
 } = actions;
 
 export default connect(
   state => ({
     preview: getPreview(state),
     selectedSource: getSelectedSource(state),
-    linesInScope: getInScopeLines(state),
```
Comment (jasonLaster) [inline src/components/Editor/Preview/index.js]: this prop is now in the action.
We can move `selectedFrameVisible` as well, which simplifies the data dependencies for the component quite a bit

File: src/utils/test-head.js [hunk 2]
```diff
@@ -78,6 +78,10 @@ function makeSymbolDeclaration(name: string, line: number) {
  */
 function waitForState(store: any, predicate: any): Promise<void> {
   return new Promise(resolve => {
+    if (predicate(store.getState())) {
+      resolve();
+    }
```
Comment (jasonLaster) [inline src/utils/test-head.js]: Not needed, but it is now consistent w/ the mochitest head helper.

**Comment by jasonLaster** [comment]
created a PR for setInScopeLines

File: src/components/Editor/Preview/index.js (hunk 1 above)
Comment (tohmjudson) [reply src/components/Editor/Preview/index.js]: It looks like you can remove `getInScopeLines` from the import (line 14) as well then

**Pull request merged**